src/components/MediaUploadZone.tsx

```diff
@@ -0,0 +1,148 @@
+import { UploadSimple, WarningCircle } from "@phosphor-icons/react";
+import { useState, useEffect, useRef } from "react";
+
+interface MediaUploadZoneProps {
+  onFilesDropped: (files: File[]) => void;
+}
+
+const MediaUploadZone: React.FC<MediaUploadZoneProps> = ({
+  onFilesDropped,
+}) => {
+  // parent ref
+  const parentRef = useRef<HTMLDivElement>(null);
+
+  // handling sizing
+  const [margin, setMargin] = useState("1em");
+  const [fontSize, setFontSize] = useState("1em");
+  // handling drop in file
+  const [isDragOver, setIsDragOver] = useState(false);
+  const [droppedFiles, setDroppedFiles] = useState<File[]>([]);
+  const [isValidFile, setIsValidFile] = useState(true);
+
+  const UploadZoneClassName =
+    "w-full h-full border-2 border-dashed flex flex-col justify-center items-center rounded-md border-primary-main " +
+    (isDragOver
+      ? " transform transition-transform duration-300 border-primary-dark ease-in-out scale-95"
+      : "");
+
+  // adjust font size according to parent ref
+  useEffect(() => {
```

Comment:
We do not need this, as this can be accomplished by using CSS only (you can use "w-full h-full" to achieve the same thing).